### PR TITLE
PLA-3758 - Add IngredientFractionsPredictor

### PR DESCRIPTION
**Pull request by dannycodes**
# Citrine Python PR

## Description 

  Add the IngredientFractionsPredictor to the
  set of available predictors in the `predictors` module.

  Add tests to both informatics and serialization test
  files.

  Document this new predictor in the docs.
  NOTE: Heavily sourced from the [Solution Design Doc](https://citrine.atlassian.net/wiki/spaces/~375952471/pages/219643922/Simple+mixture+support+on+the+Citrine+Platform#Compute-ingredient-fractions-from-a-recipe-designed)

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by dannycodes** [comment]
@latture Did my best on the `rst` file, might want to give it a read to verify correctness.

Also, the PR template has this verbiage in it:

```I have communicated the downstream consequences of the PR to others.```

I'm going to consider that a no-op for this PR.

File: src/citrine/informatics/predictors.py [hunk 1]
```diff
@@ -668,3 +668,60 @@ def _post_dump(self, data: dict) -> dict:
...
+    input_descriptor: FormulationDescriptor
+        descriptor that represents the input formulation
+    ingredients: List[str]
+
+    """
+    uid = _properties.Optional(_properties.UUID, 'id', serializable=False)
+    name = _properties.String('config.name')
+    description = _properties.String('config.description')
+    input_descriptor = _properties.Object(FormulationDescriptor, 'config.input')
+    ingredients = _properties.List(_properties.String, 'config.ingredient_names')
```
Comment (latture) [inline src/citrine/informatics/predictors.py:691]:
```suggestion
    ingredients = _properties.List(_properties.String, 'config.ingredients')
```

**Comment by latture** [comment]
@dannycodes This looks great. There are only a few minor things missing:

- We need to add the predictor to `__all__` at the top of the file.
- `IngredientsFractions` needs to be added to the type dict in `Predictor.get_type`.
- The serialization path for ingredients should be `config.ingredients` instead of `config.ingredient_names`

I added a suggestion for the latter, but we'll need to propagate the change to the serialization tests. Do you mind if I add some additional clarification in the user docs and push to this branch? I can take care of the items I listed above while I'm at it. Afterwards this should be ready to merge.

**Comment by dannycodes** [comment]
@latture patched those changes. Haven't touched the docs. Please feel free to modify them / improve them

**Comment by latture** [comment]
💯 I'll push the revisions to the docs. Then I think this is ready to merge.

**Pull request merged**